File: Weapon/ColdWeapon/Knife.php
<?php


namespace Weapon;


class Knife extends ColdWeapon
{
    private $armorPiercing;

    final public function getArmorPiercing(){
        return $this->armorPiercing;
    }
    final public function setArmorPiercing(float $armorPiercing){
        $this->armorPiercing = $armorPiercing;
    }

}

File: Weapon/ColdWeapon/Knife.py
from ColdWeapon import ColdWeapon


class Knife(ColdWeapon):
    __armorPiercing = None

    @property
    def armor_piercing(self):
        return self.__armorPiercing

    @armor_piercing.setter
    def armor_piercing(self, value: int):
        self.__armorPiercing = value
